pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -545,6 +545,14 @@ func (desc *immutable) ValidateSelf(vea catalog.ValidationErrorAccumulator) {
 		if desc.GetArrayTypeID() != descpb.InvalidID {
 			vea.Report(errors.AssertionFailedf("ALIAS type desc has array type ID %d", desc.GetArrayTypeID()))
 		}
+	case descpb.TypeDescriptor_COMPOSITE:
+		if desc.Composite == nil {
+			vea.Report(errors.AssertionFailedf("COMPOSITE type desc has nil composite type"))
+		}
+		if len(desc.Composite.TupleLabels) != len(desc.Composite.TupleContents) {
```

Comment:
I understand that we already defend on frontend that we should always have at least one tuple content, but still worth add extra validation to make sure the lengths are greater than zero.